### ISSUE TITLE
Tokenizer arguments not working

### ISSUE THREAD

**Issue by parinaya-007** [UKPLab/sentence-transformers#1432]
Tokenizer arguments aren't working when passing it to Transformers **tokenizer_args**.

```
MAX_SEQ_LEN = 32
tokenizer_params = {"padding": "max_length", "truncation": True, "max_length": MAX_SEQ_LEN}
word_embedding_model = models.Transformer(TSDAE_MODEL_BASE_PATH, 
                                          max_seq_length=MAX_SEQ_LEN,
                                          tokenizer_args=tokenizer_params)
pooling_model = models.Pooling(word_embedding_model.get_word_embedding_dimension(), 'mean')
model = SentenceTransformer(modules=[word_embedding_model, pooling_model])

input_ids = model.tokenize(["check tokens"])["input_ids"]
print(len(input_ids))
```

The output isn't equal to MAX_SEQ_LEN.

**Comment by nreimers** [comment]
Why do you want to pad always to the max length?

**Comment by dysby** [comment]
Hi @nreimers, 

I would like to revive this issue, since I want to use sbert with other variants of Roberta that need tokenization with padding to specific length, for example with Longformer, BigBird etc.

I can use transformer's API, but I think sbert could improve it's generalization if extended to support custom tokenization strategies.

**Comment by cjuracek** [comment]
Hello @nreimers - another +1 for this issue, if it's still under consideration.

I'd like to be able to truncate input to a specific length, described [here](https://huggingface.co/docs/transformers/en/pad_truncation) (truncation to specific length -> no padding).

Some options I've tried:
- `tokenizer_kwargs` argument of `SentenceTransformer.__init__()`: Only for tokenizer init args, not for tokenizer call
- `SentenceTransformer().tokenize()` does not support additional kwargs
- `SentenceTransformer().encode()` has kwargs, but not clear where they are going. Doesn't appear to have an effect on the sequence length, so I don't think this is the use case. I also cannot find documentation for this method on the [class page](https://sbert.net/docs/package_reference/sentence_transformer/SentenceTransformer.html)

Ultimately, I ended up creating a custom class inheriting from `SentenceTransformer` and overriding the `tokenize()` method to support kwargs to pass to the underlying tokenizer, but this is far from ideal.

**Comment by tomaarsen** [comment]
Hello @cjuracek,

We've kept the tokenization kind of general/without arguments as then we can easily support different forms of tokenization/processing. After all, `padding` or `truncate` arguments don't make much sense on image processing. 
Either way, I would like to improve support here, perhaps by removing the forced padding and truncation here: https://github.com/UKPLab/sentence-transformers/blob/e28f97da71f3ab910ea9cc5c078ce758295c162e/sentence_transformers/models/Transformer.py#L451-L452

And instead relying on whatever is defined in the tokenizer's configuration. Then this should also allow `tokenizer_kwargs` to start  working I believe. I'm just a bit wary that it'll backwards compatibility of model inference, which is completely unacceptable.

- Tom Aarsen

**Comment by cjuracek** [comment]
Thanks for the update @tomaarsen - appreciate the response

**Comment by drexalt** [comment]
Hey @tomaarsen !
Is it still not possible to enforce padding to max_length? One big benefit (for me anyway hehe) would be torch compiling SpladeLoss without graph breaks due to different lengths between batches. I've tested the compilation, and torch compile can knock out the intermediate logits copy, which is pretty impactful.

**Comment by tomaarsen** [comment]
Not yet, apologies. I want to look into this, but I'd like to incorporate this with a future potential refactor introducing stronger multimodality support. If I refactor the tokenization too often, I run the risk that I'll have to break backwards compatibility.

- Tom Aarsen
